### ISSUE TITLE
[wandb] partial functions don't have __name__

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#10911]
```
Traceback (most recent call last):
  File "/home/ubuntu/run_ray_tune.py", line 222, in <module>
    tune_helsinki_(args)
  File "/home/ubuntu/run_ray_tune.py", line 106, in tune_helsinki_
    ray_wandb_func = wandb_mixin(ray_func)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/tune/integration/wandb.py", line 142, in wandb_mixin
    func.__wandb_group__ = func.__name__
AttributeError: 'functools.partial' object has no attribute '__name__'
Shared connection to 34.220.26.193 closed.
Error: Command failed:
```

If I try to do: 
```
    ray_func = partial(ray_main, args)
    ray_wandb_func = wandb_mixin(ray_func)
```

**Comment by richardliaw** [comment]
cc @krfricke

**Comment by krfricke** [comment]
Closed via #10950
